src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -170,6 +171,25 @@ def ev_charger_pool(
             )
         return self._ev_charger_pools[key]
 
+    def grid(
+        self,
+    ) -> _grid.Grid | None:
```

Comment:
This is generally great, but I think `_grid.Grid` will not render correctly in the docs, you might need to do the class import like the rest (`from ._grid import Grid`). You might want to check the generated docs (`pip install -e .[dev-mkdocs]; mkdocs serve`).